### PR TITLE
macos fixes

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: LICENSE_THIRDPARTY [hunk 1]
```diff
@@ -0,0 +1,27 @@
+When building POCL for macOSX, the files lib/CL/devices/pthread/pthread_barrier.*
+are used which are licensed as below.
+
+Copyright (c) 2015, Aleksey Demakov
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
```
Comment (pjaaskel) [inline LICENSE_THIRDPARTY:7]: How does this differ from the MIT license most of the PoCL is licensed with? I don't see the difference, but IANAL.

File: lib/CL/devices/pthread/pthread_barrier.c [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+#include "pthread_barrier.h"
+
+#include <errno.h>
+
+#ifdef __APPLE__
+
+#define __unused __attribute__((unused))
+
+int
+pthread_barrierattr_init(pthread_barrierattr_t *attr __unused)
```
Comment (pjaaskel) [inline lib/CL/devices/pthread/pthread_barrier.c:36]: Please use the clang formatter script to get these files to the PoCL style.

**Review by pjaaskel — CHANGES_REQUESTED** [review]
Otherwise this barrier wrapper looks fine, and might be useful for other platforms without this pthread API.

File: LICENSE_THIRDPARTY (hunk 1 above)
Comment (jansol) [reply LICENSE_THIRDPARTY:7]: It is the [2-clause BSD license](https://github.com/ademakov/DarwinPthreadBarrier/blob/master/LICENSE). I guess having some attribution here wouldn't hurt?

**Pull request merged**